aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -99,10 +99,7 @@ inline std::vector<int64_t> construct_offsets(const at::Tensor& sizes) {
 // correct Autograd key which is AutogradNestedTensor
 c10::DispatchKeySet generate_nested_key_set(at::Tensor buffer) {
   c10::DispatchKeySet key_set =
-      (c10::DispatchKeySet(DispatchKey::NestedTensor) |
-       c10::DispatchKeySet(
-           buffer.is_cuda() ? BackendComponent::CUDABit
-                            : BackendComponent::CPUBit));
+      c10::DispatchKeySet(DispatchKey::NestedTensor) | c10::DispatchKeySet{buffer.key_set().highestBackendKey()};
```

Comment:
Spoke with Brian about this, we still assert in the constructor that the buffer_ is either CPU or Cuda but if/when other backends want to right kernels this will make that change easier